src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -89,6 +89,9 @@ class NativeCallStackStorage : public CHeapObjBase {
   }
 
   const inline NativeCallStack& get(StackIndex si) {
+    if (si._stack_index < 0 || si._stack_index >= _stacks.length()) {
+      return _fake_stack;
+    }
```

Comment:
You are right. Since the next lines check the `-1` as special case; this check should be either an `assert` or logged. I wait for @jdksjolen, as this is his code.

Reply:
Old code, delete it. No logging for this. You can always add an assert for `si._stack_index >= _stacks.length()`.